src/Xamarin.Android.Build.Tasks/Tasks/InstallApkSet.cs

```diff
@@ -37,5 +42,28 @@ internal override CommandLineBuilder GetCommandLineBuilder ()
 
 			return cmd;
 		}
+
+		const string InstallErrorRegExString = @"(?<exception>com.android.tools.build.bundletool.model.exceptions.CommandExecutionException+):(?<error>.+)";
```

Comment:
Its supposed to pick up the following from the output `com.android.tools.build.bundletool.model.exceptions.CommandExecutionException` .

Once it "detects" that it will then process everything after as part of the error. This is how the error handling works in the JavaTool base class. 

I could be using the wrong regex, I tried to copy the existing one from https://github.com/xamarin/xamarin-android/blob/main/src/Xamarin.Android.Build.Tasks/Tasks/JavaToolTask.cs#L68 . I might have got it wrong.

Reply:
@dellis1972: the `ExceptionRegExString` regex "ends with" `.+`:

```csharp
const string ExceptionRegExString = @"(?<exception>java.lang.+):(?<error>.+)";
```

Breaking it down:

 1. `(?<exception>java.lang.+)` creates a group named "exception" which starts with `java.lang` and captures everything afterward.
 2. Followed by a colon `:`
 3. `(?<error>.+)` creates a group named "error" which contains at least one character, capturing until the end of the line.

Note that both (1) and (3) use `.+`.

Compare to `InstallErrorRegExString`:

```csharp
const string InstallErrorRegExString = @"(?<exception>com.android.tools.build.bundletool.model.exceptions.CommandExecutionException+):(?<error>.+)";
```

with:

 1. `(?<exception>….CommandExecutionException+)` creates a group named "exception" which starts with `….CommandExecutionException` followed by any number of `n`s: `CommandExecutionException`, `CommandExecutionExceptionn`, `CommandExecutionExceptionnn`, `CommandExecutionExceptionnnn`…
 2. Followed by a colon `:`
 3. `(?<error>.+)` creates a group named "error" which contains at least one character, capturing until the end of the line.

(1) is what looks "wrong" to me; I don't think it needs to end with `n+`.  I believe that the `n` could be entirely removed, *or* we could consider being less restrictive, looking for anything containing `.exceptions.`:

```csharp
const string InstallErrorRegExString = @"(?<exception>.+\.exceptions\..+):(?<error>.+)";
```

How important is `CommandExecutionException`?

Reply:
> How important is `CommandExecutionException`?

It seems to be the high level exception which bundletool uses to all of its exceptions. Which is why I wanted to restrict it to this. We can try the more generate one. 
I'll just remove the `+` for now see if that works.